met-web/src/components/engagement/form/EngagementWidgets/AddContactDrawer.tsx

```diff
@@ -17,40 +17,74 @@ import { WidgetDrawerContext } from './WidgetDrawerContext';
         resolver: yupResolver(schema),
+        defaultValues: {
+            name: selectedContact ? selectedContact.name : '',
+            phone_number: selectedContact ? selectedContact.phone_number : '',
+            title: selectedContact ? selectedContact.title : '',
+            email: selectedContact ? selectedContact.email : '',
+            address: selectedContact ? selectedContact.address : '',
+            bio: selectedContact ? selectedContact.bio : '',
+        },
     });
 
+    useEffect(() => {
+        methods.setValue('name', selectedContact?.name);
+        methods.setValue('phone_number', selectedContact?.phone_number);
+        methods.setValue('title', selectedContact?.title);
+        methods.setValue('email', selectedContact?.email);
+        methods.setValue('address', selectedContact?.address);
+        methods.setValue('bio', selectedContact?.bio);
+    }, [selectedContact]);
+
     const { handleSubmit } = methods;
-    const { addContactDrawerOpen, handleAddContactDrawerOpen, loadContacts } = useContext(WidgetDrawerContext);
-    const [isCreatingContact, setIsCreatingContact] = useState(false);
 
     const onSubmit: SubmitHandler<ContactForm> = async (data: ContactForm) => {
         try {
-            setIsCreatingContact(true);
-            await postContact(data);
-            setIsCreatingContact(false);
-            handleAddContactDrawerOpen(false);
-            dispatch(openNotification({ severity: 'success', text: 'A new contact was successfully added' }));
-            loadContacts();
+            if (selectedContact) {
+                // setIsCreatingContact(true);
+                // await updateContact(data);
+                // setIsCreatingContact(false);
+                // handleAddContactDrawerOpen(false);
+                // dispatch(openNotification({ severity: 'success', text: 'Contact was successfully updated' }));
+                // loadContacts();
+            } else {
+                setIsCreatingContact(true);
+                await postContact(data);
+                setIsCreatingContact(false);
+                handleAddContactDrawerOpen(false);
+                dispatch(openNotification({ severity: 'success', text: 'A new contact was successfully added' }));
+                loadContacts();
+            }
         } catch (err) {
             console.log(err);
-            dispatch(
-                openNotification({ severity: 'error', text: 'An error occured while trying to create a new contact' }),
-            );
+            !selectedContact
+                ? dispatch(
+                      openNotification({
+                          severity: 'error',
+                          text: 'An error occured while trying to create a new contact',
+                      }),
+                  )
+                : dispatch(
+                      openNotification({ severity: 'error', text: 'An error occured while trying to update contact' }),
+                  );
```

Comment:
Maybe one message and use the term 'save' contact for both

Reply:
Ill try that

Reply:
reminding you of this